src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java

```diff
@@ -85,8 +85,13 @@ public void test(ConnectorContext connectorContext) throws ValidationException {
     config.validate(collector, oAuthInfo);
   }
 
+
   @Override
   public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    return browse(this.config);
+  }
+
+  public BrowseDetail browse(SalesforceConnectorConfig config) throws IOException {
```

Comment:
Why do you need to expose both this and constructor? Usually you either pass something in constructor or as method parameter, but not both.